### PR TITLE
Paysafe: Add gateway integration

### PR DESCRIPTION
**Pull request by meagabeth**
- Support the following AM methods: authorize, capture, purchase, verify, void, refund, store
- Add support for payment tokens in all transaction types
- Support additional methods: credit and redact
- [Paysafe card API documentation](https://developer.paysafe.com/en/cards/api/)
- [Paysafe vault API documentation](https://developer.paysafe.com/en/vault/api/)

CE-1555

Rubocop:
710 files inspected, no offenses detected
Local:
4864 tests, 74030 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
Unit:
13 tests, 44 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
Remote:
21 tests, 51 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

### CODE REVIEW

File: test/remote/gateways/remote_paysafe_test.rb [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    @credit_card = credit_card('4107857757053670')
+    @pm_token = 'Ci3S9DWyOP9CiJ5'
+    @options = {
+      billing_address: {
+        address1: '999 This Way Lane',
+        city: 'Hereville',
+        state: 'NC',
+        country: 'FR',
+        zip: '98989',
+        phone: '999-9999999'
```
Comment (naashton) [inline test/remote/gateways/remote_paysafe_test.rb:17]: Nothing wrong with defining this yourself, but you can just invoke an address via the `test_helper` by calling `address`.

File: lib/active_merchant/billing/gateways/paysafe.rb [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+      self.homepage_url = 'https://www.paysafe.com/'
+      self.display_name = 'Paysafe'
+
+      def initialize(options = {})
+        requires!(options, :username, :password, :account_id)
+        super
+      end
+
+      # Purchase requests for this gateway happen at the same endpoint as `authorize`
+      # but they include the `settleWithAuth` parameter
```
Comment (naashton) [inline lib/active_merchant/billing/gateways/paysafe.rb:20]: I don't think you need this comment, or the ones on line 51 and 59.

**Review by naashton — APPROVED** [review]
No major issues. Implementation is clean and easy to understand. Only a few nitpicks that boil down to personal preference, but otherwise I think this is good to 🚢 .

File: lib/active_merchant/billing/gateways/paysafe.rb (hunk 2 above)
Comment (meagabeth) [reply lib/active_merchant/billing/gateways/paysafe.rb:20]: Removed!

File: test/remote/gateways/remote_paysafe_test.rb (hunk 1 above)
Comment (meagabeth) [reply test/remote/gateways/remote_paysafe_test.rb:17]: Good call - Revised!

**Pull request merged**